tests/Unit/Design/pass_by_ref.cpp

```diff
@@ -11,47 +11,47 @@
 class myVecAdd {
  public:
   // CPU-side constructor. Written by the user
-  myVecAdd(Concurrency::array_view<int>& a,
-    Concurrency::array_view<int> &b,
-    Concurrency::array_view<int, 1> &c):
+  myVecAdd(hc::array_view<int>& a,
+    hc::array_view<int> &b,
+    hc::array_view<int, 1> &c):
     a_(a), b_(b), c_(c) {
   }
-  void operator() (Concurrency::index<1> idx) restrict(amp) {
+  void operator() (hc::index<1> idx) [[hc]] {
     c_[idx] = a_[idx]+b_[idx];
   }
  private:
-  Concurrency::array_view<int> a_, b_;
-  Concurrency::array_view<int>& c_;
+  hc::array_view<int> a_, b_;
+  hc::array_view<int>& c_;
 };
-void bar(void) restrict(amp,cpu) {
+void bar(void) [[cpu, hc]] {
   int* foo = reinterpret_cast<int*>(&myVecAdd::__cxxamp_trampoline);
```

Comment:
what is cxxamp_trampoline?

Reply:
It's a compiler generated entry point for the kernel (long-ish story). It's going out as well as part of the AMP removal / HC Next refactoring, but that's part of another PR.

Reply:
Okay sounds good to me